ArjunaJTS/jts/classes/com/arjuna/ats/jts/utils/Utility.java

```diff
@@ -258,6 +258,9 @@ public static final Uid getUid (org.omg.CosTransactions.Control cont)
 
 	byte[] nodeName = TxControl.getXANodeName().getBytes(StandardCharsets.UTF_8);
 
+        if (Boolean.getBoolean("GLASSFISH-21532-WORKAROUND"))
```

Comment:
I think we should still give this some more format config. We can't mandate when GF21532 is fixed and this boolean could live a long time in unmanaged space.

Add the env bean property with deprecated?